akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala

```diff
@@ -288,20 +288,28 @@ private[parser] trait CommonRules { this: Parser with StringBuilding =>
   }
 
   def `path-av` = rule {
-    ignoreCase("path=") ~ OWS ~ `path-value` ~> { (c: HttpCookie, pathValue: String) => c.copy(path = Some(pathValue)) }
+    ignoreCase("path=") ~ OWS ~ `path-value` ~> { (c: HttpCookie, pathValue: String) => c.withPath(pathValue) }
   }
 
   // http://www.rfc-editor.org/errata_search.php?rfc=6265
   def `path-value` = rule {
     capture(zeroOrMore(`av-octet`)) ~ OWS
   }
 
+  def `same-site-av` = rule {
+    ignoreCase("samesite=") ~ OWS ~ `same-site-value` ~> { (c: HttpCookie, sameSiteValue: String) => c.withSameSite(sameSite = SameSite(sameSiteValue)) }
+  }
+
+  def `same-site-value` = rule {
+    capture(ignoreCase("lax") | ignoreCase("strict") | ignoreCase("none")) ~ OWS
```

Comment:
This means `samesite=wrong` will fail to parse. Do we want that?

Reply:
Would be ok I think for the first version. If people run into that problem often enough, we can think about providing alternatives.

Reply:
Also, since it is the server-side sending those, only our client-side would be affected where cookie usage currently isn't as important.

Reply:
Ah, you're right. Let's keep it like this then. I'll update the tests - do you think we should call this out in the release/migration notes?

Reply:
Wouldn't hurt to mention it. It's not really worse than before, though.

Reply:
Good point, let's leave it like this.